### PR TITLE
[lldb] Treat user aliases the same as built-ins when tab completing

### PR DESCRIPTION
**Pull request by DavidSpickett**
Previously we would check all built-ins first for suggestions,
then check built-ins and aliases. This meant that if you had
an alias brkpt -> breakpoint, "br" would complete to "breakpoint".

Instead of giving you the choice of "brkpt" or "breakpoint".

### CODE REVIEW

**Comment by DavidSpickett** [comment]
This is a PR "stacked" on https://github.com/llvm/llvm-project/pull/65973, so the first 2 commits are from that and the original one. Review with the commits tab instead of "files changed".

File: lldb/test/API/functionalities/abbreviation/TestAbbreviations.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def test_command_abbreviations_and_aliases(self):
         self.assertTrue(result.Succeeded())
         self.assertEqual("apropos script", result.GetOutput())
 
-        command_interpreter.ResolveCommand("h", result)
+        command_interpreter.ResolveCommand("he", result)
```
Comment (jimingham) [inline lldb/test/API/functionalities/abbreviation/TestAbbreviations.py:23]: Why did you need to change "h" to "he" here?  "help" is the only command that begins with "h" so "h" should complete to "help".

**Review by jimingham — COMMENTED** [review]
This looks right to me, but I'm worried about why you had to change the "h" completion test to "he"?

File: lldb/test/API/functionalities/abbreviation/TestAbbreviations.py (hunk 1 above)
Comment (DavidSpickett) [reply lldb/test/API/functionalities/abbreviation/TestAbbreviations.py:23]:
```
$ ./bin/lldb
(lldb) h
Ambiguous command 'h'. Possible matches:
        help
        history
<...>
'history' is an abbreviation for 'session history'
```

I figured if we're treating aliases and abreviations and the like all the same, then changing this test was the logical thing to do. Unless abbreviations should never resolve to aliases?

Comment (DavidSpickett) [reply lldb/test/API/functionalities/abbreviation/TestAbbreviations.py:23]: Though even if I was right, I should explain that in the test case.

Comment (jimingham) [reply lldb/test/API/functionalities/abbreviation/TestAbbreviations.py:24]: But there's an explicit alias from "h" to "help".  So "h" shouldn't be ambiguous, and it isn't on current TOT lldb.  If this change is making `h` ambiguous, then that's not right.

Comment (jimingham) [reply lldb/test/API/functionalities/abbreviation/TestAbbreviations.py:24]: Huh, I don't see where we explicitly alias "help" to "h", but in the current TOT lldb (well, as of last Friday) and in the latest Apple release, "h" is not ambiguous:

(lldb) version
lldb version 18.0.0git (git@github.com:llvm/llvm-project.git revision 04a6dc24db333ee8bfdf50791a18024f30086ff6)
  clang revision 04a6dc24db333ee8bfdf50791a18024f30086ff6
  llvm revision 04a6dc24db333ee8bfdf50791a18024f30086ff6
(lldb) h
Debugger commands:
  apropos           -- List debugger commands related to a word or subject.
  breakpoint        -- Commands for operating on breakpoints (see 'help b' for

I wonder what's different.  Whatever we do, `h` really should mean "help" that's a far more common usage.

Comment (DavidSpickett) [reply lldb/test/API/functionalities/abbreviation/TestAbbreviations.py:24]: I don't see it in `CommandInterpreter::Initialize` with the rest but you're right there's definitely something in there, h -> help on Linux too.

Comment (DavidSpickett) [reply lldb/test/API/functionalities/abbreviation/TestAbbreviations.py]: I figured it out. Previously, when you start with "h":
* exact match commands - fails
* exact match aliases - fails
* inexact match commands - succeeds

And we would skip looking in the aliases for an inexact match. That's why "history" never shows up.

Now I'm treating aliases and commands on the same level, when the inexact match is looked for both show up.

I've added an explicit alias h -> help to fix this, and restored the test to what it was.

Comment (DavidSpickett) [reply lldb/test/API/functionalities/abbreviation/TestAbbreviations.py]: So it never was an intentional alias because it never needed aliasing, now it does.

**Comment by jimingham** [comment]
I don't see where you approve a PR in this UI but this is approved.
